packages/web-components/src/components/card/card.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2019, 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement } from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import { BASIC_COLOR_SCHEME } from '../../globals/shared-enums';
+import DDSLink from '../../globals/internal/link';
+import DDSCardFooter from './card-footer';
+import styles from './card.scss';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * The table mapping slot name with the private property name that indicates the existence of the slot content.
+ */
+const slotExistencePropertyNames = {
+  eyebrow: '_hasEyebrow',
+  heading: '_hasHeading',
+};
+
+/**
+ * Card.
+ * @element dds-card
+ */
+@customElement(`${ddsPrefix}-card`)
+class DDSCard extends DDSLink {
+  /**
+   * `true` if there is eyebrow content.
+   */
+  private _hasEyeblow;
+
+  /**
+   * `true` if there is heading content.
+   */
+  private _hasHeading;
+
+  /**
+   * `true` if there is copy content.
+   */
+  private _hasCopy;
+
+  /**
+   * Handles `slotchange` event.
+   */
+  private _handleSlotChange({ target }: Event) {
```

Comment:
In the spirit of the slack conversation we just had, wondering if we should consider moving any of the inner methods here out?

Reply:
@jeffchew Do you have more details, e.g. what the code will be like and the motivation to do so?

Reply:
more about being able to test internal methods as isolated functions. So this means either making some methods non-private so that unit tests can do isolated tests around that method, or taking out the method from the class entirely and export/create unit tests around that.

Reply:
Given the `private` runs in trasnspiler level (rather than [browser run-time level like `#handleSlotChange`](https://github.com/tc39/proposal-private-fields/blob/master/OLD_README.md)), we can still access to the private field by `(cardInstance as any)._handleSlotChange`.

Reply:
@jeffchew as web components uses classes, so it's not that of a problem, in opposite to our react functional components

Reply:
makes sense!